src/context/FoxEthProvider/FoxEthProvider.tsx

```diff
@@ -0,0 +1,297 @@
+          const data = await getOpportunityData({
+            contractAddress: opportunity.contractAddress,
+            ethAssetPrecision: ethAsset.precision,
+            ethPrice: ethMarketData.price,
+            lpAssetPrecision,
+            address: connectedWalletEthAddress,
+            foxPrice: foxMarketData.price,
+            foxAssetPrecision,
+          })
+          if (!data) return opportunity
+          const { tvl, apr, balances, expired } = data
+          return {
+            ...opportunity,
+            cryptoAmount: balances.cryptoBalance,
+            fiatAmount: balances.fiatBalance,
+            isLoaded: isLpAprLoaded,
+            unclaimedRewards: balances.unclaimedRewards,
+            expired,
+            apy: lpApr
+              ? bnOrZero(apr)
+                  .plus(lpApr ?? 0)
+                  .toString()
+              : undefined,
+            tvl,
+          }
+        }),
+      )
+      const totalOpBalances = newOpportunities.reduce(
+        (acc, cur) => acc.plus(bnOrZero(cur.fiatAmount)),
+        bnOrZero(0),
+      )
+      setFoxFarmingTotalBalance(totalOpBalances.toFixed(2))
+      setFoxFarmingOpportunities(newOpportunities)
+    } catch (error) {
+      console.error('error', error)
+    } finally {
+      setFarmingLoading(false)
+    }
+  }, [
+    connectedWalletEthAddress,
+    foxFarmingOpportunities,
+    ethAsset.precision,
+    ethMarketData.price,
+    lpAssetPrecision,
+    foxMarketData.price,
+    foxAssetPrecision,
+    isLpAprLoaded,
+    lpApr,
+  ])
+
+  const reloadLpOpportunity = useCallback(async () => {
+    try {
+      const balances = await calculateHoldings()
+      if (balances) {
+        const { lpBalance, foxBalance, ethBalance } = balances
+        setLpFoxBalance(foxBalance.toString())
+        setLpEthBalance(ethBalance.toString())
+        const totalLpBalance = bnOrZero(lpBalance).div(`1e${lpAssetPrecision}`)
+        const ethValue = ethBalance.times(ethMarketData.price)
+        const foxValue = foxBalance.times(foxMarketData.price)
+        const fiatAmount = ethValue.plus(foxValue).toFixed(2)
+        const totalSupply = await getLpTVL()
+        setFoxEthLpOpportunity({
+          ...lpOpportunity,
+          cryptoAmount: totalLpBalance?.toString() ?? '0',
+          fiatAmount,
+          isLoaded: true,
+          apy: lpApr ?? undefined,
+          tvl: totalSupply ?? '',
+        })
+      }
+    } catch (error) {
+      console.error('error', error)
+    } finally {
+      setLpLoading(false)
+    }
+  }, [
+    calculateHoldings,
+    lpApr,
+    getLpTVL,
+    foxMarketData.price,
+    ethMarketData.price,
+    lpAssetPrecision,
+  ])
+
+  // reload opportunities when wallet changes
+  useEffect(() => {
+    if (!(connectedWalletEthAddress && lpApr)) return
+    ;(async () => {
+      reloadFarmingOpportunities()
+      reloadLpOpportunity()
+    })()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
If we need this functionally, comment out why, else add all deps

Reply:
comment added.